docs/features/techdocs/using-cloud-storage.md

```diff
@@ -309,3 +310,77 @@ and read the static generated documentation files. When you start the backend of
 the app, you should be able to see
 `techdocs info Successfully connected to the Azure Blob Storage container` in
 the logs.
+
+## Configuring OpenStack Swift Container with TechDocs
+
+Follow the
+[official OpenStack Api documentation](https://docs.openstack.org/api-ref/identity/v3/)
+for the latest instructions on the following steps involving Open Stack Storage.
+
+**1. Set `techdocs.publisher.type` config in your `app-config.yaml`**
+
+Set `techdocs.publisher.type` to `'openStackSwift'`.
+
+```yaml
+techdocs:
+  publisher:
+    type: 'openStackSwift'
+```
+
+**2. Create an OpenStack Swift Storage Container**
+
+Create a dedicated container for TechDocs sites.
+[Refer to the official documentation](https://docs.openstack.org/mitaka/user-guide/dashboard_manage_containers.html).
+
+TechDocs will publish documentation to this container and will fetch files from
+here to serve documentation in Backstage. Note that the container names are
+globally unique.
+
+Set the config `techdocs.publisher.openStackSwift.containerName` in your
+`app-config.yaml` to the name of the container you just created.
+
+```yaml
+techdocs:
+  publisher:
+    type: 'openStackSwift'
+    openStackSwift:
+      containerName: 'name-of-techdocs-storage-container'
+```
+
+**3. Authentication using app-config.yaml**
+
+Set the configs in your `app-config.yaml` to the your container name.
+
+https://docs.openstack.org/api-ref/identity/v3/?expanded=password-authentication-with-unscoped-authorization-detail#password-authentication-with-unscoped-authorization
+for more details.
+
+```yaml
+techdocs:
+  publisher:
+    type: 'openStackSwift'
+    openStackSwift:
+      containerName: 'name-of-techdocs-storage-bucket'
+      credentials:
+        userName:
+          $env: OPENSTACK_SWIFT_STORAGE_USERNAME
+        password:
+          $env: OPENSTACK_SWIFT_STORAGE_PASSWORD
+      authUrl:
+        $env: OPENSTACK_SWIFT_STORAGE_AUTH_URL
+      keystoneAuthVersion:
+        $env: OPENSTACK_SWIFT_STORAGE_AUTH_VERSION
```

Comment:
Think some of these could be left out from being specified using `$env` keys. I'm assuming it's very rare that the keystone version changes between deployments, so prolly just put the value straight in the config file instead

Reply:
We think the same thing `keystoneAuthVersion`, `domainId` and `domainName` optional values. You can see defaults here, https://github.com/erdoganoksuz/backstage/blob/d679b32f20f3771eced12488f244d27df7f4832d/packages/techdocs-common/src/stages/publish/openStackSwift.ts#L65

We don't know how to say these values optional in documentation if you give a suggestion we do this.

Maybe like that;
`keystoneAuthVersion?:`

Or
`keystoneAuthVersion: (optional)`

Reply:
Yeah I think `keystoneAuthVersion: (optional)` works, and then perhaps only inline an example auth URL?

Reply:
Agreed that we don't need to show the usage of `$env` for `keystoneAuthVersion`, `domainId` and `domainName`. Let's write inline docs like `keystoneAuthVersion: (optional, default: 'v3')`

`authUrl` can also be an inline string without `$env`.